drivers/rtc/rtc-wbec.c

```diff
@@ -0,0 +1,271 @@
+// SPDX-License-Identifier: GPL-2.0-only
+/*
+ * TODO
+ * Write description
+ */
+
+#include <linux/module.h>
+#include <linux/kernel.h>
+#include <linux/init.h>
+#include <linux/i2c.h>
+#include <linux/rtc.h>
+#include <linux/platform_device.h>
+#include <linux/time.h>
+#include <linux/regmap.h>
+#include <linux/of_device.h>
+#include <linux/mfd/wbec.h>
+
+#define WBEC_RTC_ALARM_FLAG_POLL_PERIOD_MS		500
+
+struct wbec_rtc_config {
+	struct regmap_config regmap;
+};
+
+struct wbec_rtc {
+	struct rtc_device	*rtc;
+	struct regmap		*regmap;
+};
+
+static int wbec_rtc_read_time(struct device *dev, struct rtc_time *tm)
+{
+	struct wbec_rtc *wbec_rtc = dev_get_drvdata(dev);
+
+	int rc;
+	u16 regs[4];
+
+	/*
+	 * while reading, the time/date registers are blocked and not updated
+	 * anymore until the access is finished. To not lose a second
+	 * event, the access must be finished within one second. So, read all
+	 * time/date registers in one turn.
+	 */
+	rc = regmap_bulk_read(wbec_rtc->regmap, WBEC_REG_RTC_TIME_SECS_MINS, regs,
+				  ARRAY_SIZE(regs));
+	if (rc)
+		return rc;
+
+	/* if the clock has lost its power it makes no sense to use its time */
+	// TODO Check power loss?
+
+
+	tm->tm_sec = regs[0] & 0x00FF;
+	tm->tm_min = regs[0] >> 8;
+	tm->tm_hour = regs[1] & 0x00FF;
+	tm->tm_mday = regs[1] >> 8;
+	tm->tm_wday = regs[2] & 0x00FF;
+	tm->tm_mon = regs[2] >> 8;
+	tm->tm_year = regs[3];
+	tm->tm_year += 100;
+
+	return 0;
+}
+
+static int wbec_rtc_set_time(struct device *dev, struct rtc_time *tm)
+{
+	struct wbec_rtc *wbec_rtc = dev_get_drvdata(dev);
+	int rc;
+	u16 regs[4];
+
+	// TODO Remove debug
+	dev_info(dev, "%s function\n", __func__);
+
+	/* hours, minutes and seconds */
+	regs[0] = tm->tm_sec;
+	regs[0] |= tm->tm_min << 8;
+	regs[1] = tm->tm_hour;
+
+	/* Day of month, 1 - 31 */
+	regs[1] |= tm->tm_mday << 8;
+
+	/* Day, 0 - 6 */
+	regs[2] = tm->tm_wday & 0x07;
+
+	/* month, 1 - 12 */
+	regs[2] = (tm->tm_mon + 1) << 8;
```

Comment:
Кажется, тут только что потеряли день недели
```suggestion
	regs[2] |= (tm->tm_mon + 1) << 8;
```